speech/speechManager.py

```diff
@@ -1,89 +1,84 @@
-    data = provider.getSpeakData(text, voiceId)
-    if config.cacheData and data and len(data) > 0:
-        util.saveCacheData(text, providerId, voiceId, data)
-    return data
-
-def isSpeaking():
-    for provider in speechProviders.values():
-        if hasattr(provider, "isSpeaking") and provider.isSpeaking():
-            return True
-    return False
-
-def stop():
-    for provider in speechProviders.values():
-        if hasattr(provider, "stop"):
-            provider.stop()
-
-def getVoices():
-    allVoices = []
-    for provider in config.speechProviderList:
-        voices = provider.getVoices()
-        for voice in voices:
-            voice["providerId"] = provider.getProviderId()
-            voice["type"] = provider.getVoiceType()
-            voice["name"] = voice["name"] + ", " + provider.getProviderId()
-            allVoices.append(voice)
-
-    return allVoices
-
-def initProviders():
-    for provider in config.speechProviderList:
-        id = provider.getProviderId() if hasattr(provider, "getProviderId") else "fill_provider_id"
-        if id == "fill_provider_id":
-            raise Exception("ERROR: '{}' is an invalid provider ID!".format(id))
-
-        if id in speechProviders:
-            raise Exception("ERROR: duplicated speech provider with ID '{}'!".format(id))
-
-        for fnName in requiredFnsAll:
-            if not hasattr(provider, fnName):
-                raise Exception("ERROR: speech provider '{}' doesn't implement function '{}'!".format(id, fnName))
-
-        voiceType = provider.getVoiceType()
-        additionalRequiredFns = None
-        if voiceType == constants.VOICE_TYPE_EXTERNAL_PLAYING:
-            additionalRequiredFns = requiredFnsPlaying
-        elif voiceType == constants.VOICE_TYPE_EXTERNAL_DATA:
-            additionalRequiredFns = requiredFnsData
-        else:
-            raise Exception("ERROR: voice type of provider '{}' invalid (must be '{}' or '{}')!".format(id, constants.VOICE_TYPE_EXTERNAL_PLAYING, constants.VOICE_TYPE_EXTERNAL_DATA))
-
-        for fnName in additionalRequiredFns:
-            if not hasattr(provider, fnName):
-                raise Exception("ERROR: speech provider '{}' doesn't implement function '{}'!".format(id, fnName))
-
-        voices = provider.getVoices()
-        for voice in voices:
-            name = voice["id"] or voice["name"] or "noVoiceName"
-            for key in requiredVoiceKeys:
-                if key not in voice:
-                    raise Exception("ERROR: voice '{}' of provider '{}' doesn't have required key '{}'!".format(name, id, key))
-
-        speechProviders[id] = provider # store in map
-
-initProviders()
+from typing import Optional, List, Dict, Any
+from provider_platform_data import provider as platform_provider
+from provider_test_data import provider as test_provider
+
+# List of all available providers
+providers = [
```

Comment:
before I've used `config.speechProviderList` in this file, now the providers are defined directly here. So I assume either `config.speechProviderList` isn't needed anymore or the providers should be defined in `config.py` as before and used here instead of the local variable `providers`?! The reason why I've moved it to `config.py` was to have everything which can be configured at a single place.